core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -159,6 +181,13 @@ String getPreferredIp(List<String> preferredTypes) {
    */
   void forceRefresh() {
     synchronized (instanceDataGuard) {
+      // Don't force a refresh until the current forceRefresh operation
+      // has produced a successful refresh.
+      if (forceRefreshRunning) {
+        return;
+      }
+
+      forceRefreshRunning = true;
       nextInstanceData.cancel(false);
       if (nextInstanceData.isCancelled()) {
```

Comment:
Shall we remove this if block and always schedule a next refresh since we also always call cancel? This after having observed that cancel doesn't always work as we might think.